src/clib/pio_file.c

```diff
@@ -109,6 +109,7 @@ int PIOc_createfile(const int iosysid, int *ncidp, int *iotype,
         return PIO_ENOMEM;
 
     /* Fill in some file values. */
+    file->fh = -1;
```

Comment:
Note that logging statements in the code already assume (or don't make sense otherwise) that this handle is initialized.